File: src/test/java/com/ject/studytrip/trip/fixture/DailyGoalFixture.java
package com.ject.studytrip.trip.fixture;

import com.ject.studytrip.trip.domain.factory.DailyGoalFactory;
import com.ject.studytrip.trip.domain.model.DailyGoal;
import com.ject.studytrip.trip.domain.model.Trip;
import org.springframework.test.util.ReflectionTestUtils;

public class DailyGoalFixture {
    private static final String DEFAULT_TITLE = "TEST TITLE";

    public static DailyGoal createDailyGoal(Trip trip) {
        return DailyGoalFactory.create(trip, DEFAULT_TITLE);
    }

    public static DailyGoal createDailyGoalWithId(Long id, Trip trip) {
        DailyGoal dailyGoal = DailyGoalFactory.create(trip, DEFAULT_TITLE);
        ReflectionTestUtils.setField(dailyGoal, "id", id);

        return dailyGoal;
    }
}

File: src/test/kotlin/com/ject/studytrip/trip/fixture/DailyGoalFixture.kt
package com.ject.studytrip.trip.fixture

import com.ject.studytrip.trip.domain.factory.DailyGoalFactory
import com.ject.studytrip.trip.domain.model.DailyGoal
import com.ject.studytrip.trip.domain.model.Trip
import org.springframework.test.util.ReflectionTestUtils

class DailyGoalFixture(
    private val trip: Trip,
    private val title: String = "TEST 데일리 목표 제목",
) {
    fun create(): DailyGoal = DailyGoalFactory.create(trip, title)

    fun createWithId(id: Long): DailyGoal =
        create().also {
            ReflectionTestUtils.setField(it, "id", id)
        }
}
